# Entendiendo los datos de CCS

In [20]:
### Librerias

import pandas as pd

In [21]:
# Dataframe Fact Ventas

df_fv = pd.read_csv('./data_source/data_query_for_fac_ventas.csv')

In [22]:
df_fv.head(3)

,NUMERO_DE_IDENTIFICACION,PRIMER_APELLIDO,SEGUNDO_APELLIDO,PRIMER_NOMBRE,SEGUNDO_NOMBRE,CURSO,PRECIO_NETO,FECHA_DE_PAGO,RENOVACION,FECHA_DE_VENTA,PROFESION,GENERO,FECHA_DE_NACIMIENTO,CIUDAD_REGION,MODALIDAD,PROCEDENCIA,MEDIO_DE_PAGO,RESPONSABLE_VENTA,elaboro
0,98430412,Correa,Cuero,Ivan,Hernando,Ove,50000.0,2024-04-01,No,2024-04-01,Tecnólogo en atención prehospitalaria,Masculino,1978-02-26,Pasto - Nariño,Virtual asincrónica,Facebook,Bancolombia,Lucio Francisco Rosero Muñoz,Sebastian Rosero Patiño
1,98430412,Correa,Cuero,Ivan,Hernando,Primeros auxilios,50000.0,2024-04-01,No,2024-04-01,Tecnólogo en atención prehospitalaria,Masculino,1978-02-26,Pasto - Nariño,Virtual asincrónica,Facebook,Bancolombia,Lucio Francisco Rosero Muñoz,Sebastian Rosero Patiño
2,30746429,Martinez,Tabla,Martha,Isabel,Violencia sexual,30000.0,2024-04-05,No,2024-04-01,Odontólogo general,Femenino,1970-06-07,Pasto - Nariño,Virtual asincrónica,Facebook,Nequi,Provired,Sebastian Rosero Patiño


In [23]:
# Informacion general del Dataframe
df_fv.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7701 entries, 0 to 7700
Data columns (total 19 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   NUMERO_DE_IDENTIFICACION  7701 non-null   int64  
 1   PRIMER_APELLIDO           7701 non-null   object 
 2   SEGUNDO_APELLIDO          7566 non-null   object 
 3   PRIMER_NOMBRE             7701 non-null   object 
 4   SEGUNDO_NOMBRE            7328 non-null   object 
 5   CURSO                     7700 non-null   object 
 6   PRECIO_NETO               7701 non-null   float64
 7   FECHA_DE_PAGO             7701 non-null   object 
 8   RENOVACION                7468 non-null   object 
 9   FECHA_DE_VENTA            7701 non-null   object 
 10  PROFESION                 5905 non-null   object 
 11  GENERO                    5802 non-null   object 
 12  FECHA_DE_NACIMIENTO       7698 non-null   object 
 13  CIUDAD_REGION             7527 non-null   object 
 14  MODALIDA

In [24]:
# Dimenciones del Dataframe

df_fv.shape

(7701, 19)

In [25]:
df_fv.columns.to_list()

['NUMERO_DE_IDENTIFICACION',
 'PRIMER_APELLIDO',
 'SEGUNDO_APELLIDO',
 'PRIMER_NOMBRE',
 'SEGUNDO_NOMBRE',
 'CURSO',
 'PRECIO_NETO',
 'FECHA_DE_PAGO',
 'RENOVACION',
 'FECHA_DE_VENTA',
 'PROFESION',
 'GENERO',
 'FECHA_DE_NACIMIENTO',
 'CIUDAD_REGION',
 'MODALIDAD',
 'PROCEDENCIA',
 'MEDIO_DE_PAGO',
 'RESPONSABLE_VENTA',
 'elaboro']

In [26]:
# Tipos de datos del Dataframe
df_fv.dtypes

NUMERO_DE_IDENTIFICACION      int64
PRIMER_APELLIDO              object
SEGUNDO_APELLIDO             object
PRIMER_NOMBRE                object
SEGUNDO_NOMBRE               object
CURSO                        object
PRECIO_NETO                 float64
FECHA_DE_PAGO                object
RENOVACION                   object
FECHA_DE_VENTA               object
PROFESION                    object
GENERO                       object
FECHA_DE_NACIMIENTO          object
CIUDAD_REGION                object
MODALIDAD                    object
PROCEDENCIA                  object
MEDIO_DE_PAGO                object
RESPONSABLE_VENTA            object
elaboro                      object
dtype: object

In [27]:
# Analisis estadistico rapido

df_fv.describe()

,NUMERO_DE_IDENTIFICACION,PRECIO_NETO
count,7.701000e+03,7701.000000
mean,7.814038e+08,31963.900160
std,4.700817e+08,23394.154644
min,5.101770e+05,0.000000
25%,8.706522e+07,25000.000000
50%,1.085261e+09,30000.000000
75%,1.086300e+09,32000.000000
max,2.000023e+09,280000.000000


#### CCS Data frame, descripcion

La base de datos de ventas llamada fact ventas de CCS cuenta con 7701 registros y 19 columnas.

De las columnas presentada aquella que no pueden tener datos nulos (null) son:
- NUMERO_DE_IDENTIFICACION 
- PRIMER_APELLIDO
- PRIMER_NOMBRE
- CURSO
- PRECIO_NETO
- FECHA_DE_PAGO
- FECHA_DE_VENTA

Segun la inspeccion de valores nulos hay otras columnas que si tienes valores faltantes estos valores deben ser tratados
desntro la limpieza de datos a fin de normalizar dichos valores faltantes.
